src/sqlfluff/core/rules/std.py

```diff
@@ -1276,13 +1225,6 @@ def _eval(self, segment, parent_stack, raw_stack, **kwargs):
                     insert_str = ""
                     init_pos = segment.segments[0].pos_marker
 
-                    # Add intial whitespace if we need to...
-                    if raw_stack[-1].name not in ["whitespace", "newline"]:
-                        insert_buff.append(
-                            self.make_whitespace(raw=" ", pos_marker=init_pos)
-                        )
-                        insert_str += " "
-
```

Comment:
Hmmmmm. I wonder if we still need this (or something that achieves the same). The case that it covers would be the following syntax which I've checked is valid:

```
select foo.bar from (select 1 as bar)foo
```

Reply:
Ah thank you! For some reason I was just blanking on this...